src/thermo/MixtureFugacityTP.cpp

```diff
@@ -810,10 +785,244 @@ void MixtureFugacityTP::_updateReferenceStateThermo() const
+        } 
+        disc = 0.0;
+    }
+
+    if (disc < -1e-14) {
+        // disc<0 then we have three distinct roots.
+        nSolnValues = 3;
+    } else if (fabs(disc) < 1e-14) {
+        // disc=0 then we have two distinct roots (third one is repeated root)
+        nSolnValues = 2;
+        // We are here as p goes to zero.
+    } else if (disc > 1e-14) {
+        // disc> 0 then we have one real root.
+        nSolnValues = 1;
+    }
+
+    double tmp;
+    // One real root -> have to determine whether gas or liquid is the root
+    if (disc > 0.0) {
+        double tmpD = sqrt(disc);
+        double tmp1 = (- yN + tmpD) / (2.0 * an);
+        double sgn1 = 1.0;
+        if (tmp1 < 0.0) {
+            sgn1 = -1.0;
+            tmp1 = -tmp1;
+        }
+        double tmp2 = (- yN - tmpD) / (2.0 * an);
+        double sgn2 = 1.0;
+        if (tmp2 < 0.0) {
+            sgn2 = -1.0;
+            tmp2 = -tmp2;
+        }
+        double p1 = pow(tmp1, 1./3.);
+        double p2 = pow(tmp2, 1./3.);
+        double alpha = xN + sgn1 * p1 + sgn2 * p2;
+        Vroot[0] = alpha;
+        Vroot[1] = 0.0;
+        Vroot[2] = 0.0;
+    } else if (disc < 0.0) {
+        // Three real roots alpha, beta, gamma are obtained.
+        double val = acos(-yN / h);
+        double theta = val / 3.0;
+        double twoThirdPi = 2. * Pi / 3.;
+        double alpha = xN + 2. * delta * cos(theta);
+        double beta = xN + 2. * delta * cos(theta + twoThirdPi);
+        double gamma = xN + 2. * delta * cos(theta + 2.0 * twoThirdPi);
+        Vroot[0] = beta;
+        Vroot[1] = gamma;
+        Vroot[2] = alpha;
+
+        for (int i = 0; i < 3; i++) {
+            tmp = an * Vroot[i] * Vroot[i] * Vroot[i] + bn * Vroot[i] * Vroot[i] + cn * Vroot[i] + dn;
+            if (fabs(tmp) > 1.0E-4) {
+                for (int j = 0; j < 3; j++) {
+                    if (j != i && fabs(Vroot[i] - Vroot[j]) < 1.0E-4 * (fabs(Vroot[i]) + fabs(Vroot[j]))) {
+                        writelog("MixtureFugacityTP::solveCubic(T = {}, p = {}):"
+                                 " WARNING roots have merged: {}, {}\n",
+                                 T, pres, Vroot[i], Vroot[j]);
+                    }
+                }
+            }
+        }
+    } else if (disc == 0.0) {
+        //Three equal roots are obtained, i.e. alpha = beta = gamma
+        if (yN < 1e-18 && h < 1e-18) {
+            // yN = 0.0 and h = 0 i.e. disc = 0
+            Vroot[0] = xN;
+            Vroot[1] = xN;
+            Vroot[2] = xN;
+        } else {
+            // h and yN need to figure out whether delta^3 is positive or negative
+            if (yN > 0.0) {
+                tmp = pow(yN/(2*an), 1./3.);
+                // In this case, tmp and delta must be equal.
+                if (fabs(tmp - delta) > 1.0E-9) {
+                    throw CanteraError("MixtureFugacityTP::solveCubic", "Inconsistancy in cubic solver : solver is poorly conditioned.");
```

Comment:
Two instances of this:
```suggestion
                    throw CanteraError("MixtureFugacityTP::solveCubic", "Inconsistency in cubic solver : solver is poorly conditioned.");
```
I'd also suggest line wrapping some of these error messages. They're way beyond our target line length of 80 characters.

Reply:
Corrected.